Wikipedia/Code/ArticlePlaceGroupViewController.swift

```diff
@@ -0,0 +1,332 @@
+                return
+            }
+        }
+        
+        guard !didDeselect else {
+            return
+        }
+        
+        delegate?.articlePlaceGroupViewControllerDidDismiss(self)
+    }
+    
+    func selectPlaceView(_ placeView: ArticlePlaceView) {
+        guard !isHidden else {
+            return
+        }
+        for placeView in placeViews {
+            guard placeView.isSelected else {
+                continue
+            }
+            placeView.setSelected(false, animated: true)
+            delegate?.articlePlaceGroupViewController(self, didDeselectPlaceView: placeView)
+        }
+        guard placeView.reuseIdentifier != "zoom" else {
+            delegate?.articlePlaceGroupViewControllerDidSelectZoom(self)
+            return
+        }
+        placeView.setSelected(true, animated: true)
+        delegate?.articlePlaceGroupViewController(self, didSelectPlaceView: placeView)
+    }
+    
+    func layoutPlaceViews(center: CGPoint) {
+        let count = placeViews.count
+        var i = 1
+        for placeView in placeViews {
+            let theta = 2*M_PI / Double(count)
+            let angle = -1*M_PI_2 + theta*Double(i - 1)
+            let x = radius * cos(angle)
+            let y = radius * sin(angle)
+            placeView.center = CGPoint(x: center.x + CGFloat(x), y: center.y + CGFloat(y))
+            placeView.transform = CGAffineTransform.identity
+            i += 1
+        }
+        
+        groupView.center = center
+        zoomPlaceView.center = center
+    }
+    
+    func layout(center: CGPoint) {
+        layoutPlaceViews(center: center)
+        drawGradient(center: center)
+    }
+    
+    var center: CGPoint?
+    
+    func drawGradient(center: CGPoint) {
+        let scale = traitCollection.displayScale
+        UIGraphicsBeginImageContextWithOptions(view.bounds.size, false, scale)
+        guard let ctx = UIGraphicsGetCurrentContext() else {
+            return
+        }
+        let maxDimension = max(view.bounds.size.width, view.bounds.size.height)
+        let grayColor = UIColor(white: 0, alpha: 0.5).cgColor
+        let clearColor = UIColor.clear.cgColor
+        let colorSpace = CGColorSpaceCreateDeviceRGB()
+        let locations: [CGFloat] = [0, CGFloat(radius)/maxDimension, 1]
+        guard let gradient = CGGradient(colorsSpace: colorSpace, colors: [clearColor, grayColor, grayColor] as CFArray, locations: locations) else {
+            return
+        }
+        ctx.drawRadialGradient(gradient, startCenter: center, startRadius: 10, endCenter: center, endRadius: maxDimension, options: [])
+        let image = UIGraphicsGetImageFromCurrentImageContext()
+        tintView.image = image
+        UIGraphicsEndImageContext()
+    }
+    
+    func show(center: CGPoint) {
+        self.center = center
+       
+        drawGradient(center: center)
+        
+        let rotationTransform = CGAffineTransform(rotationAngle: CGFloat(-0.99*M_PI))
```

Comment:
To force it to rotate in a certain direction. Exactly M_PI and negative M_PI don't go the same way, this way it forces clockwise out and counter-clockwise back in.